File: producer-consumer-BE/src/main/java/com/example/producerconsumerbe/Service/ProducerConsumerService.java
package com.example.producerconsumerbe.Service;

import com.example.producerconsumerbe.Service.Model.Machine;
import com.example.producerconsumerbe.Service.Model.MyQueue;
import com.example.producerconsumerbe.Service.Model.Network;
import com.example.producerconsumerbe.Service.Model.Product;
import com.example.producerconsumerbe.Service.Socket.WebSocketService;
import org.springframework.stereotype.Service;

@Service
public class ProducerConsumerService {

    private Network network;
    private final WebSocketService webSocketService;

    public ProducerConsumerService(WebSocketService webSocketService){
        network = new Network(webSocketService);
        this.webSocketService = webSocketService;
    }

    public void addMachine(){
        network.addMachine(new Machine(webSocketService));
    }

    public void addQueue(){
        network.addQueue(new MyQueue());
    }

    public void addProduct(){
        network.addProduct(new Product());
    }

    public void connectMachineToQueue(String machineId,String queueId){
        network.connectMachineToQueue(machineId,queueId);
    }

    public void connectQueueToMachine(String machineId,String queueId){
        network.connectQueueToMachine(machineId,queueId);
    }

    public void pause(String queueId){
        network.pause(queueId);
    }

    public void start(){
        network.start();
    }

    public void replay(){
        network.replay();
    }

    public void clear() {
        network = new Network(webSocketService);
    }

}

File: Producer-Consumer-FE/src/app/producer-consumer.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";

@Injectable({
  providedIn: 'root'
})
export class ProducerConsumerService {

  private URL: string = "http://localhost:9090/api/producer_consumer/";

  constructor(private http: HttpClient) {
  }

  public addMachine() {
    this.http.post(this.URL + "addMachine", {}).subscribe();
  }

  public addQueue() {
    this.http.post(this.URL + "addQueue", {}).subscribe();
  }

  public addProduct() {
    this.http.post(this.URL + "addProduct", {}).subscribe();
  }

  public connectMachineToQueue(machineId: string, queueId: string) {
    this.http.post(this.URL + "connectMtoQ", {
      "machine_id": machineId,
      "queue_id": queueId
    }).subscribe();
  }

  public connectQueueToMachine(machineId: string, queueId: string) {
    this.http.post(this.URL + "connectQtoM", {
      "machine_id": machineId,
      "queue_id": queueId
    }).subscribe();
  }

  public start() {
    this.http.post(this.URL + "start", {}).subscribe();
  }

  public pause(id : string) {
    this.http.post(this.URL + "pause", {
      "queue_id": id,
    }).subscribe();
  }

  public replay() {
    this.http.post(this.URL + "replay", {}).subscribe();
  }

  public clear(){
    this.http.post(this.URL + "clear" , {}).subscribe();
  }

}
